### PR TITLE
Melgeek mj63

### PR DESCRIPTION
**Pull request by melgeek001365**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by ridingqwerty — APPROVED** [review]

**Comment by melgeek001365** [comment]
@fauxpark @drashna @noroadsleft  please check it, Thank you very much! :)

**Review by fauxpark — APPROVED** [review]

File: keyboards/melgeek/mj63/mj63.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define XXX KC_NO
+
+#define LAYOUT_60_ansi( \
```
Comment (noroadsleft) [inline keyboards/melgeek/mj63/mj63.h:23]: This isn't a `LAYOUT_60_ansi`, but it is a fairly common layout.

```suggestion
#define LAYOUT_63_ansi( \
```

File: keyboards/melgeek/mj63/info.json [hunk 2]
```diff
@@ -0,0 +1,80 @@
+{
+    "keyboard_name": "MJ63",
+    "url": "",
+    "maintainer": "melgeek001365",
+    "width": 15,
+    "height": 5,
+    "layouts": {
+        "LAYOUT_60_ansi": {
```
Comment (noroadsleft) [inline keyboards/melgeek/mj63/info.json:8]:
```suggestion
        "LAYOUT_63_ansi": {
```

File: keyboards/melgeek/mj63/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_60_ansi(
```
Comment (noroadsleft) [inline keyboards/melgeek/mj63/keymaps/default/keymap.c:20]:
```suggestion
    [0] = LAYOUT_63_ansi(
```

File: keyboards/melgeek/mj63/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        KC_GESC,   KC_1,     KC_2,     KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,     KC_9,    KC_0,    KC_MINS,         KC_EQL,  KC_BSPC,
+        KC_TAB,    KC_Q,     KC_W,     KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,     KC_O,    KC_P,    KC_LBRC,         KC_RBRC, KC_BSLS,
+        KC_CAPS,   KC_A,     KC_S,     KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,     KC_L,    KC_SCLN, KC_QUOT,                  KC_ENT,
+        KC_LSFT,             KC_Z,     KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,     KC_COMM, KC_DOT,  RSFT_T(KC_SLSH), KC_UP,   KC_DEL,
+        KC_LCTL,   KC_LGUI,  KC_LALT,                    KC_SPC,                              MO(1),   MO(2),   KC_LEFT,         KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT_60_ansi(
```
Comment (noroadsleft) [inline keyboards/melgeek/mj63/keymaps/default/keymap.c:27]:
```suggestion
    [1] = LAYOUT_63_ansi(
```

File: keyboards/melgeek/mj63/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        KC_GRV,    KC_F1,    KC_F2,    KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,    KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,
+        _______,   _______,  _______,  _______, _______, _______, KC_CALC, _______, KC_INS,   _______, KC_PSCR, KC_SLCK, KC_PAUS, RESET,
+        _______,   _______,  _______,  _______, _______, _______, _______, _______, _______,  _______, KC_HOME, KC_PGUP,          EEP_RST,
+        _______,             _______,  _______, _______, _______, _______, NK_TOGG, _______,  _______, KC_END,  KC_PGDN, KC_VOLU, KC_MUTE,
+        _______,   _______,  _______,                    _______,                             _______, _______, KC_MPRV, KC_VOLD, KC_MNXT
+    ),
+    [2] = LAYOUT_60_ansi(
```
Comment (noroadsleft) [inline keyboards/melgeek/mj63/keymaps/default/keymap.c:34]:
```suggestion
    [2] = LAYOUT_63_ansi(
```

File: keyboards/melgeek/mj63/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_60_ansi(
```
Comment (noroadsleft) [inline keyboards/melgeek/mj63/keymaps/via/keymap.c:20]:
```suggestion
    [0] = LAYOUT_63_ansi(
```

File: keyboards/melgeek/mj63/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        KC_GESC,   KC_1,     KC_2,     KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,     KC_9,    KC_0,    KC_MINS,         KC_EQL,  KC_BSPC,
+        KC_TAB,    KC_Q,     KC_W,     KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,     KC_O,    KC_P,    KC_LBRC,         KC_RBRC, KC_BSLS,
+        KC_CAPS,   KC_A,     KC_S,     KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,     KC_L,    KC_SCLN, KC_QUOT,                  KC_ENT,
+        KC_LSFT,             KC_Z,     KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,     KC_COMM, KC_DOT,  RSFT_T(KC_SLSH), KC_UP,   KC_DEL,
+        KC_LCTL,   KC_LGUI,  KC_LALT,                    KC_SPC,                              MO(1),   MO(2),   KC_LEFT,         KC_DOWN, KC_RGHT
+    ),
+    [1] = LAYOUT_60_ansi(
```
Comment (noroadsleft) [inline keyboards/melgeek/mj63/keymaps/via/keymap.c:27]:
```suggestion
    [1] = LAYOUT_63_ansi(
```

File: keyboards/melgeek/mj63/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+        KC_GRV,    KC_F1,    KC_F2,    KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,    KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,
+        _______,   _______,  _______,  _______, _______, _______, KC_CALC, _______, KC_INS,   _______, KC_PSCR, KC_SLCK, KC_PAUS, RESET,
+        _______,   _______,  _______,  _______, _______, _______, _______, _______, _______,  _______, KC_HOME, KC_PGUP,          EEP_RST,
+        _______,             _______,  _______, _______, _______, _______, NK_TOGG, _______,  _______, KC_END,  KC_PGDN, KC_VOLU, KC_MUTE,
+        _______,   _______,  _______,                    _______,                             _______, _______, KC_MPRV, KC_VOLD, KC_MNXT
+    ),
+    [2] = LAYOUT_60_ansi(
```
Comment (noroadsleft) [inline keyboards/melgeek/mj63/keymaps/via/keymap.c:34]:
```suggestion
    [2] = LAYOUT_63_ansi(
```

File: keyboards/melgeek/mj63/keymaps/via/keymap.c [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+        _______,   _______,  _______,  _______, _______, _______, _______, _______, _______,  _______, _______, _______, _______, _______,
+        _______,   RGB_TOG,  _______,  RGB_HUI, RGB_HUD, RGB_SAI, RGB_SAD, RGB_VAI, RGB_VAD,  RGB_MOD, _______, _______, _______, _______,
+        _______,   _______,  _______,  _______, _______, _______, _______, _______, RGB_SPI,  RGB_SPD, _______, _______,          _______,
+        _______,             _______,  _______, _______, _______, _______, _______, _______,  _______, _______, _______, _______, _______,
+        _______,   _______,  _______,                    _______,                             _______, _______, _______, _______, _______
+    ),
+    [3] = LAYOUT_60_ansi(
```
Comment (noroadsleft) [inline keyboards/melgeek/mj63/keymaps/via/keymap.c:41]:
```suggestion
    [3] = LAYOUT_63_ansi(
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/10424/files).

File: keyboards/melgeek/mj63/rev1/rules.mk [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+COMMAND_ENABLE = no            # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no          # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes              # USB Nkey Rollover
+BACKLIGHT_ENABLE = no          # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no           # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no          # Enable Bluetooth
+RGB_MATRIX_ENABLE = IS31FL3741 # Use RGB matrix
+NO_USB_STARTUP_CHECK = no      # Disable initialization only when usb is plugged in
```
Comment (noroadsleft) [inline keyboards/melgeek/mj63/rev1/rules.mk:20]:
```suggestion
NO_USB_STARTUP_CHECK = no      # Disable initialization only when usb is plugged in

LAYOUTS = 60_ansi_arrow
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/melgeek/mj63/mj63.c [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+    suspend_wakeup_init_user();
+}
+
+void keyboard_pre_init_kb(void) {
+    setPinOutput(A2);
+    writePinHigh(A2);
+    keyboard_pre_init_user();
+}
+
+#endif
```
Comment (zvecr) [inline keyboards/melgeek/mj63/mj63.c:128]: Given this ties to the config within `keyboards/melgeek/mj63/rev1/config.h`, this should be moved to keyboards/melgeek/mj63/rev1/rev1.c.

```suggestion
```

File: keyboards/melgeek/mj63/rev1/config.h [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
```
Comment (zvecr) [inline keyboards/melgeek/mj63/rev1/config.h:20]: You have a via keymap included in this PR, and they generally don't like this ID. Changing it within their guidelines is one of the quickest ways forward https://caniusevia.com/docs/configuring_qmk#changes-to-keyboard-directorys-configh.

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/melgeek/mj63/mj63.c (hunk 11 above)
Comment (melgeek001365) [reply keyboards/melgeek/mj63/mj63.c:128]: > Given this ties to the config within `keyboards/melgeek/mj63/rev1/config.h`, this should be moved to keyboards/melgeek/mj63/rev1/rev1.c.
Thank you for your advice. but I think it's not good time to move the content of mj63.c and mj63.h to rev1.c rev1.h now, cause We can't predict what will change in the next version. maybe there are few changes between this version and the next version. so most of the code is duplicated.

Comment (zvecr) [reply keyboards/melgeek/mj63/mj63.c:128]: Its about aligning the logic, you cant have generic led config at the top level, while having some specific stuff at the revision level. What you have now is just inconsistent. And as you state, you dont know what the next revision might use. Thus if you change to a different led driver, the config would most likely have to change.

File: keyboards/melgeek/mj63/rev1/config.h (hunk 12 above)
Comment (melgeek001365) [reply keyboards/melgeek/mj63/rev1/config.h]: @zvecr please check it Thank you

**Review by drashna — APPROVED** [review]
Travis CI failure is due to a community layout (@mrsendyyk), and is not a blocker for this PR.

LGTM

**Pull request merged**